lua/wire/zvm/zvm_opcodes.lua

```diff
@@ -407,47 +411,87 @@ ZVM.OpcodeTable[59] = function(self)  --MOD
 end
 --------------------------------------------------------------------------------
 ZVM.OpcodeTable[60] = function(self)  --BIT
-  self:Dyn_Emit("$L BITS = VM:IntegerToBinary($1)")
-  self:Dyn_Emit("VM.CMPR = BITS[math.floor($2)] or 0")
-  self:Dyn_Emit("VM.TMR = VM.TMR + 30")
+  self:Dyn_Emit("if VM.IPREC > 32 then")
+    self:Dyn_Emit("$L BITS = VM:IntegerToBinary($1)")
+    self:Dyn_Emit("VM.CMPR = BITS[math.floor($2)] or 0")
+    self:Dyn_Emit("VM.TMR = VM.TMR + 30")
+  self:Dyn_Emit("else")
+    self:Dyn_Emit("if bit.band($1,math.pow(2,$2)) ~= 0 then")
```

Comment:
`math.ldexp(1,x) might be faster than math.pow(2,x)`

Reply:
> `math.ldexp(1,x)` might be faster than `math.pow(2,x)`

It may have a lua error on non integer in the exp, d'you think it'd still be faster even if I had to floor exp on every call/make a function that wraps ldexp and floors the exp?

Reply:
some of those look like they should be floored, yes

Reply:
IPREC can (for some reason) be a non-integer if set by the user, so I'm going to guarantee it gets floored on set rather than floor on use, then we can safely use ldexp over pow